mesa/extra.py

```diff
@@ -56,11 +63,11 @@ def at_most(agentset, n, rng_istance):
         return list(rng_istance.choice(agentset, n, replace=False))
 
 def weighted_n_of(n, agentset, weight_function, rng_istance):
-    # todo: check for positives
     p = [float(weight_function(x)) for x in agentset]
+    if any([True for i in p if i < 0]):
+        min_value = min(p)
+        p = [i-min_value for i in p]
```

Comment:
In case there are negative weights. In a subsequent PR, I added a check in case they were all 0.